### PR TITLE
Remove isInf, isNan

### PR DESCRIPTION
**Pull request by dneto0**
Fixes: #2270

### CODE REVIEW

File: wgsl/index.bs [hunk 1]
```diff
@@ -8494,19 +8494,11 @@ See [[#function-calls]].
   <thead>
     <tr><th>Parameterization<th>Overload<th>Description
   </thead>
-  <tr algorithm="isNan">
-  <td rowspan=4>
+  <tr algorithm="isFinite">
```
Comment (alan-baker) [inline wgsl/index.bs:8497]: Shouldn't we remove isFinite too?

Comment (dneto0) [reply wgsl/index.bs:8497]: Discussed in #2309. 

We have agreement on isNan and isFinite. I'd rather land this and then do a second MR for isFinite.

**Review by alan-baker — APPROVED** [review]

**Comment by almarklein** [comment]
In one of our shaders (in pygfx) there can be nans in the input data (storage buffer). We need to detect these nans and make the shader behave differently if a nan is found. Right now this works (code [here](
https://github.com/pygfx/pygfx/blob/22ab65925cfb19540b2dbb953d875ba4e9dce8a2/pygfx/data_files/shaders/line.wgsl#L66-L74 )):
```wgsl
    let is_nan = min(w, 1.0) == 1.0 && max(w, -1.0) == -1.0
```

We have two questions related to this:

1) what would be the recommended way to check for nans in input data (no math applied yet).

2) what happens with nans in the input data in a fastmath implementation? Would it result in bogus data?

3) I read that on Metal fastmath is on by default. Is this also the case with wgpu/naga?

PS: our use-case is rendering lines, and allowing users to insert gaps by using nans, which is a common trick in dataviz.

**Comment by almarklein** [comment]
Thanks for the answers @ben-clayton!

**Pull request merged**